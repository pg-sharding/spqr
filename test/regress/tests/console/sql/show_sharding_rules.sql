CREATE SHARDING RULE rule1 COLUMNS id;
CREATE SHARDING RULE rule2 COLUMNS idx;
CREATE SHARDING RULE rule3 COLUMNS idy;

SHOW sharding_rules;

DROP DISTRIBUTION ALL CASCADE;
DROP SHARDING RULE ALL;
DROP KEY RANGE ALL;