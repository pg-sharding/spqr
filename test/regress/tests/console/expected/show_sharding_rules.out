
		SQPR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE SHARDING RULE rule1 COLUMNS id;
                                        add sharding rule                                        
-------------------------------------------------------------------------------------------------
 created sharding rule rule1 for table (*) with columns [id, hash: x->x] in default distribution
(1 row)

CREATE SHARDING RULE rule2 COLUMNS idx;
                                        add sharding rule                                         
--------------------------------------------------------------------------------------------------
 created sharding rule rule2 for table (*) with columns [idx, hash: x->x] in default distribution
(1 row)

CREATE SHARDING RULE rule3 COLUMNS idy;
                                        add sharding rule                                         
--------------------------------------------------------------------------------------------------
 created sharding rule rule3 for table (*) with columns [idy, hash: x->x] in default distribution
(1 row)

SHOW sharding_rules;
 Sharding Rule ID | Distribution ID | Table Name | Columns | Hash Function 
------------------+-----------------+------------+---------+---------------
 rule1            | default         | *          | id      | x->x
 rule2            | default         | *          | idx     | x->x
 rule3            | default         | *          | idy     | x->x
(3 rows)

DROP DISTRIBUTION ALL CASCADE;
 drop distribution 
-------------------
(0 rows)

DROP SHARDING RULE ALL;
   drop sharding rule   
------------------------
 dropped sharding rule 
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

